bootloader/firmware/updater.c

```diff
@@ -0,0 +1,99 @@
+#include "platform_specific_hal_conf.h"
+#include "bootloader/core/updater.h"
+#include "bootloader/core/util.h"
+#include "bootloader/firmware/constants.h"
+
```

Comment:
Why is this file a .c file instead of a .cpp file? 
Is it because this is a HAL and it needs to be in c?

Reply:
In order to keep the binary small, all the bootloader code is to be written in C.

Reply:
Dumb question. Why does the bootloader need to be small? Is it because it is actually stored on the physical memory of the device and we want to take up as little memory as possible?

Reply:
correct

Reply:
The smaller the bootloader, the more space there is for the main applcation.